eng/common/scripts/Helpers/Metadata-Helpers.ps1

```diff
@@ -80,3 +80,54 @@ function GetPrimaryCodeOwner ([string]$TargetDirectory)
     Write-Warning "No code owner found in $TargetDirectory."
     return $null
 }
+
+function GetDocsMsService($clientPackageInfo, $mgmtPackageInfo, $serviceName) 
+{
+  $service = $serviceName.ToLower().Replace(' ', '').Replace('/', '-')
+  if ($clientPackageInfo -and $clientPackageInfo[0].MSDocService) {
+    # Use MSDocService in csv metadata to override the service directory
+    # TODO: Use taxonomy for service name -- https://github.com/Azure/azure-sdk-tools/issues/1442
+    $service = $clientPackageInfo[0].MSDocService
+  }
+  elseif ($mgmtPackageInfo -and $mgmtPackageInfo[0].MSDocService) {
+    $service = $mgmtPackageInfo[0].MSDocService
+  }
+  Write-Host "The service of package: $service"
+  return $service
+}
+
+function GenerateDocsMsMetadata($language, $serviceName, $tenantId, $clientId, $clientSecret, $msService) 
+{
+  $langTitle = "Azure $serviceName SDK for $language"
+  $langDescription = "Reference for Azure $serviceName SDK for $language"
+  # Github url for source code: e.g. https://github.com/Azure/azure-sdk-for-js
+  $serviceBaseName = $serviceName.ToLower().Replace(' ', '').Replace('/', '-')
+  $author = GetPrimaryCodeOwner -TargetDirectory "/sdk/$serviceBaseName/"
+  $msauthor = ""
+  if (!$author) {
+    LogError "Cannot fetch the author from CODEOWNER file."
```

Comment:
Missing author exits the whole script?

Reply:
Return is better here.

Reply:
I'm not certain I agree with this. This effectively makes headers `""` when we can't assign an owner. Given that we set more metadata than just the owner, I think we should log the error, then return the header with a default author.

With the way it is now, we will simply multiple the # of warnings on the docs.ms build side. Because now we will be missing _all_ the necessary front-matter, (service, language, date, etc)